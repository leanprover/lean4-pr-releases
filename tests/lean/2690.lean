/-!
# Ensure that inductive types/structures whose fields all live in `Prop` also live in `Prop`
-/

inductive Foo₁ (α : Type _) where
  | foo : (∀ a : α, a = a) → Foo₁ α

#check Foo₁

structure Foo₂ (α : Type _) where
  h : ∀ a : α, a = a

#check Foo₂

