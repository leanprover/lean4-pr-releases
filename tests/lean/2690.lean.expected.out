Foo₁.{u_1} (α : Type u_1) : Prop
Foo₂.{u_1} (α : Type u_1) : Prop
